asynctest/case.py

```diff
@@ -347,6 +347,13 @@ def debug(self):
             self._unset_loop()
 
     def _run_test_method(self, method):
+        # If the coroutine is decorated by hypothesis wrap
+        # it before running.
+        if getattr(method, "is_hypothesis_test", False) and asyncio.iscoroutinefunction(
+            method.hypothesis.inner_test
+        ):
+            method.hypothesis.inner_test = wrap_in_sync(method.hypothesis.inner_test)
```

Comment:
What's is the type of the method when `is_hypothesis_test` is true? Is it something that any of the `inspect.iscoroutinefunction` / `inspect.iscoroutine` / `inspect.isawaitable` functions recognize?

Reply:
It is just a normal function/callable. Hypothesis `given` decorator will "swallow" the coroutine and replace it with a normal function.

Reply:
And what would be the result of `method()` then? A coroutine / awaitable?

Reply:
If I recall correct it will be `None` and a [python warning](https://docs.python.org/3/library/asyncio-dev.html#asyncio-coroutine-not-scheduled) when debug is true.